### PR TITLE
Remove obsolete apps manifest preview custom header

### PR DESCRIPTION
**Pull request by chmouel**
Apps manifest are now out of preview and uses mediaTypeV3:

https://docs.github.com/en/rest/reference/apps#create-a-github-app-from-a-manifest

### CODE REVIEW

File: github/apps_manifest.go
Comment (gmlewis) [inline github/apps_manifest.go:40]: Please remove this line, as this header is added to all requests here:
https://github.com/google/go-github/blob/master/github/github.go#L383

File: github/apps_manifest_test.go
Comment (gmlewis) [inline github/apps_manifest_test.go:33]: Let's please also remove this line for conformity with the rest of the repo.

**Review by gmlewis — APPROVED** [review]
Thank you, @chmouel !
LGTM.

Awaiting second LGTM before merging.

**Comment by gmlewis** [comment]
For future PRs, please do not use force push in this repo. See CONTRIBUTING.md for details. Thanks.

**Review by wesleimp — APPROVED** [review]
💙

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Comment by bluekeyes** [comment]
@gmlewis thanks for checking, I think this is probably fine. The GitHub docs for the supported enterprise versions don't reference this header (not that it's a great guarantee the header is unused) and I've personally never used this endpoint with the instances I maintain.

**Pull request merged**
